src/internal/connector/onedrive/collection.go

```diff
@@ -317,12 +323,15 @@ type itemAndAPIGetter interface {
 // downloadContent attempts to fetch the item content.  If the content url
 // is expired (ie, returns a 401), it re-fetches the item to get a new download
 // url and tries again.
-func downloadContent(
+func (oc *Collection) downloadContent(
```

Comment:
As with the other comment, I don't see this as being sensibly owned by the Collection itself.  It makes more sense, and will likely be more conveniently portable and testable, if it remains a function, rather than a method, and extends its parameters to include the url cache.